nimble/host/src/ble_sm.c

```diff
@@ -1723,6 +1723,20 @@ ble_sm_pair_exec(struct ble_sm_proc *proc, struct ble_sm_result *res,
     }
 }
 
+static bool
+ble_sm_verify_auth_requirements(uint8_t authreq)
+{
+    /* For now we check only SC only mode. I.e.: when remote request
+     * for legacy pairing, make sure legacy pairing is supported.
```

Comment:
I'd reword this since remote cannot *request* legacy pairing, it just indicates whether it support LE SC - if it does not support LE SC, we need to support legacy pairing and this is what we check here.